autoPyTorch/pipeline/components/setup/network_embedding/base_network_embedding.py

```diff
@@ -31,22 +31,21 @@ def transform(self, X: Dict[str, Any]) -> Dict[str, Any]:
     def build_embedding(self, num_input_features: np.ndarray, num_numerical_features: int) -> nn.Module:
         raise NotImplementedError
 
-    def _get_args(self, X: Dict[str, Any]) -> Tuple[None, None]:  # Tuple[int, np.ndarray]:
+    def _get_args(self, X: Dict[str, Any]) -> Tuple[int, np.ndarray]:
         # Feature preprocessors can alter numerical columns
-        # if len(X['dataset_properties']['numerical_columns']) == 0:
-        #     num_numerical_columns = 0
-        # else:
-        #     X_train = copy.deepcopy(X['backend'].load_datamanager().train_tensors[0][:2])
-        #
-        #     numerical_column_transformer = X['tabular_transformer'].preprocessor. \
-        #         named_transformers_['numerical_pipeline']
-        #     num_numerical_columns = numerical_column_transformer.transform(
-        #         X_train[:, X['dataset_properties']['numerical_columns']]).shape[1]
-        # num_input_features = np.zeros((num_numerical_columns + len(X['dataset_properties']['categorical_columns'])),
-        #                               dtype=int)
-        # categories = X['dataset_properties']['categories']
-        #
-        # for i, category in enumerate(categories):
-        #     num_input_features[num_numerical_columns + i, ] = len(category)
-        # return num_numerical_columns, num_input_features
-        return None, None
+        if len(X['dataset_properties']['numerical_columns']) == 0:
+            num_numerical_columns = 0
+        else:
+            X_train = copy.deepcopy(X['backend'].load_datamanager().train_tensors[0][:2])
```

Comment:
We do not have to deepcopy `X_train` because we do not change `X_train`?
Also it is not intuitive what `train_tensors[0][:2]` implies, so could you add the shape of `X_train` as a comment-out?

Reply:
So, in this node we are checking what the output shape would be for the training dataset based on the preprocessing applied to it. Again, this is not something new to `AutoPyTorch` as this is the same way it has been defined in development, see [here](https://github.com/automl/Auto-PyTorch/blob/06e67de5017b4cccad9398e24a3d9f0bd8176da3/autoPyTorch/pipeline/components/setup/network_embedding/base_network_embedding.py#L37-L40)

Reply:
We do perform feature preprocessing.

Reply:
yeah it is only called once per configuration, when we are fitting the embedding layer.